streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2328,582 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+                "    Processor: KTABLE-MAPVALUES-0000000010 (stores: [store])\n" +
+                "      --> KTABLE-TOSTREAM-0000000011\n" +
+                "      <-- KSTREAM-TOTABLE-0000000009\n" +
+                "    Processor: KTABLE-TOSTREAM-0000000011 (stores: [])\n" +
+                "      --> KSTREAM-SINK-0000000012\n" +
+                "      <-- KTABLE-MAPVALUES-0000000010\n" +
+                "    Sink: KSTREAM-SINK-0000000012 (topic: output)\n" +
+                "      <-- KTABLE-TOSTREAM-0000000011\n\n"));
+
+        try (
+            final TopologyTestDriver driver = new TopologyTestDriver(
+                topology,
+                mkProperties(mkMap(
+                    mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy"),
+                    mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "test"),
+                    mkEntry(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory("kafka-test").getAbsolutePath())
+                )),
+                Instant.ofEpochMilli(0L))) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, new StringSerializer(), new StringSerializer(), Instant.ofEpochMilli(0L), Duration.ZERO);
+            final TestOutputTopic<String, Integer> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.Integer().deserializer());
+            final KeyValueStore<String, String> store = driver.getKeyValueStore(storeName);
+            
+            inputTopic.pipeInput("A", "green", 10L);
+            inputTopic.pipeInput("B", "green", 9L);
+            inputTopic.pipeInput("A", "blue", 12L);
+            inputTopic.pipeInput("C", "yellow", 15L);
+            inputTopic.pipeInput("D", "green", 11L);
+
+            final Map<String, Integer> expectedStore = new HashMap<>();
+            expectedStore.putIfAbsent("A", 1);
+            expectedStore.putIfAbsent("B", 6);
+            expectedStore.putIfAbsent("C", 24);
+            expectedStore.putIfAbsent("D", 6);
+
+            assertEquals(expectedStore, asMap(store));
+
+            assertEquals(
+                asList(
+                    new TestRecord<>("A", 6, Instant.ofEpochMilli(10)),
+                    new TestRecord<>("B", 6, Instant.ofEpochMilli(9)),
+                    new TestRecord<>("A", 1, Instant.ofEpochMilli(12)),
+                    new TestRecord<>("C", 24, Instant.ofEpochMilli(15)),
+                    new TestRecord<>("D", 6, Instant.ofEpochMilli(11))),
+                outputTopic.readRecordsToList());
+
+        }
+    }
+
+    private static Map<String, String> asMap(final KeyValueStore<String, String> store) {
+        final HashMap<String, String> result = new HashMap<>();
+        store.all().forEachRemaining(kv -> result.put(kv.key, kv.value));
+        return result;
+    }
+
+    @Test
+    public void shouldPropagateRepartitionFlagAfterKTableJoin() {
```

Comment:
The name of this test does not match what we expect to happen. We have a `map().toTable()` that will repartition the data, and we have a `selectKey().join()` that will also repartition directly.

Note that in the existing test (that you removed above) `shouldPropagateRepartitionFlagAfterGlobalKTableJoin` a global-ktable is used -- for that case, repartitioning should happen after the join -- however, when we use a regular `KTable` that is created via `toTable()` repartitioning must happen before the join (and the topology descriptions shows that both repartitionings happens correctly before the join).

Overall, I have the impression that this test is redundant?